mycroft/skills/settings.py

```diff
@@ -94,7 +94,7 @@ def __init__(self, directory, name):
         self._settings_path = join(directory, 'settings.json')
         self._meta_path = join(directory, 'settingsmeta.json')
         self.is_alive = True
-        self.loaded_hash = hash(str(self))
+        self.loaded_hash = hash(frozenset(self.items()))
```

Comment:
Since this will fail with recursive dicts, what if we use `hash(json.dumps(self, sort_keys=True))`?

Also, make sure to keep this consistent with the method used below in `self._is_stored()`.

Reply:
I noticed the issue with recursive dicts but obviously failed to fix all of them. Sorted json.dumps should be the cleanest solution. Will switch over to that.